anchor/client/src/config.rs

```diff
@@ -127,21 +128,21 @@ pub fn from_cli(cli_args: &Node) -> Result<Config, String> {
             .map_err(|e| format!("Failed to create {:?}: {:?}", config.data_dir, e))?;
     }
 
-    if let Some(beacon_nodes) = &cli_args.beacon_nodes {
-        config.beacon_nodes = beacon_nodes
-            .iter()
-            .map(|s| SensitiveUrl::parse(s))
-            .collect::<Result<_, _>>()
-            .map_err(|e| format!("Unable to parse beacon node URL: {:?}", e))?;
-    }
-
-    if let Some(execution_nodes) = &cli_args.execution_nodes {
-        config.execution_nodes = execution_nodes
-            .iter()
-            .map(|s| SensitiveUrl::parse(s))
-            .collect::<Result<_, _>>()
-            .map_err(|e| format!("Unable to parse execution node URL: {:?}", e))?;
-    }
+    read_urls(
+        &mut config.beacon_nodes,
+        &cli_args.beacon_nodes,
+        "beacon node",
+    )?;
```

Comment:
```suggestion
if let Some(ref beacon_nodes) = cli_args.beacon_nodes {
    read_urls(&mut config.beacon_nodes, beacon_nodes, "beacon node")?;
}
```

Reply:
Imo readability of the code is also important. It's more intuitive to not call the function if the input is none.

Reply:
Another option is grouping the configs in something like this:

```rust
pub enum NodeType {
    Beacon,
    Proposer,
    Execution(ExecutionType),
}

pub enum ExecutionType {
    Http,
    WebSocket,
}

pub struct NodeEndpoint {
    pub node_type: NodeType,
    pub url: SensitiveUrl,
}

pub struct NodesConfig {
    pub endpoints: Vec<NodeEndpoint>,
}
```

But I don't know if it'd make other things more difficult

Reply:
Or even simpler

```rust
#[derive(Debug)]
pub enum Node {
    Beacon(SensitiveUrl),
    Proposer(SensitiveUrl),
    ExecutionHttp(SensitiveUrl),
    ExecutionWebSocket(SensitiveUrl),
}

#[derive(Debug)]
pub struct NodesConfig {
    pub nodes: Vec<Node>,
}
```

Reply:
It's not that bad to add a `get_beacon_nodes` function to NodesConfig, this `vec` is pretty small. But I see you don't like it, and I don't have a strong opinion at all.